subt/radio.py

```diff
@@ -33,52 +35,55 @@ def draw_positions(arr):
 class Radio(Node):
     def __init__(self, config, bus):
         super().__init__(config, bus)
-        bus.register('radio', 'artf_xyz', 'breadcrumb')
-        self.min_transmit_dt = timedelta(seconds=10)  # TODO config?
+        bus.register('radio', 'artf_xyz', 'breadcrumb', 'robot_xyz')
+        self.min_transmit_dt = 0  # i.e. every sim_time_sec -> every simulated second
         self.last_transmit = None
+        self.sim_time_sec = None
         self.recent_packets = []
         self.verbose = config.get('verbose', False)
         self.debug_robot_poses = []
+        self.message_counter = 0
 
-    def send_data(self, data):
-        self.last_transmit = self.publish('radio', data + b'\n')
-        return self.last_transmit
+    def send_data(self, channel, data):
+        raw = serialize([self.message_counter, channel, data])
```

Comment:
`data` can be `numpy` but we do not support payload that is part python native and part numpy

Reply:
? I do not understand? Is there a problem if we send depth data for example (yeah, really small depth image).

Reply:
we do not support message of type `[int, numpy.ndarray]` - it needs to be either all python types or numpy ndarray - you cannot put ndarray inside a list.

Reply:
OK, I understand - so it is our current limitation of OSGAR serialization. As we do not need (yet) to send the numpy data I would leave it (and review it when the need will come).

Reply:
That is actually not a limitation but a design decision.